src/Cache/Engine/ApcEngine_CLASS_ALIAS_RENAME_BEFORE_MERGE.php

```diff
@@ -0,0 +1,18 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         1.2.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Cache\Engine;
+
+// @deprecated Add backwards compat alias.
+class_alias('Cake\Cache\Engine\ApcuEngine', 'Cake\Cache\Engine\ApcEngine');
```

Comment:
I don't think this file is supposed to still be here.

Reply:
Sure? If peeos specify ApcEngine then this alias will kick in?

Should be named ‘ApcEngine.php’ but then the guthub diff of ApcEngine>ApcuEngine ain’t work for review purposes.

Reply:
Should this be 'apcu' so it loads ApcuEngine?
https://github.com/cakephp/cakephp/pull/11313/commits/ac0fb72059696420c2231bd66bfd33094c128ad4#diff-4d9f8cefbaed0af0a66e03a6206f6e3bR79

Reply:
@inoas Did you notice the filename? `src/Cache/Engine/ApcEngine_CLASS_ALIAS_RENAME_BEFORE_MERGE.php`

Reply:
Yes I did @saeideng really wanted to keep the diff view on github alive. The moment I rename that file to `ApcEngine.php` that moment the github diff rendering from ApcEngine.php <-> ApcuEngine.php will break and it will show ApcuEngine.php as a new file.

The file exists because I think we should add a class alias until 4.x hits in case people used ApcEngine directly for BC reasons.

Reply:
The rename was unnecessary, brings no benefit. We can rename it in 4.0.

Reply:
Well I asked before...
https://github.com/cakephp/cakephp/pull/11279#discussion_r143287747
But sure I can rename back everything.
I am not sure that it hurts in any way now.

Reply:
> The moment I rename that file to ApcEngine.php that moment the github diff rendering from ApcEngine.php <-> ApcuEngine.php will break and it will show ApcuEngine.php as a new file.

That's expected, can't do anything about it. We are definitely not going to keep filename `ApcEngine_CLASS_ALIAS_RENAME_BEFORE_MERGE.php`.

Reply:
Obviously I am going to change that class name before merge :trollface: - the rename is there just to make reviewing better!

Reply:
Inoas
I mean about keeping old file history into new file
Not  old file seperatly
So remove certainly 
`e_CLASS_ALIAS_RENAME_BEFORE_MERGE`

Reply:
So and then we don't have an alias file at all :)?
BC?
Before merge I would rename this file to ApcEngine.php.
* [ ] Good?
* [ ] Bad?

Reply:
>So and then we don't have an alias file at all :)?

no

>BC?

no related to BC :)

> Before merge I would rename this file to ApcEngine.php.

rename now
tracing change is good for new file not for old file that you remove all lines

Reply:
GITHUB show new file correctly

Reply:
Okay you don't believe me. I will rename the file and the diff-view on github will not work anymore :)

Reply:
I now understand your mean :)